### PR TITLE
Make Device more explicit.

### PR DESCRIPTION
**Pull request by pschuh**
This should help construct the tensor on the right device for better xla-tensor support.

### CODE REVIEW

File: Sources/TensorFlow/Layers/Core.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ public struct Flatten<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
 @frozen
 public struct Reshape<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
   /// The target shape.
-  @noDerivative public let shape: Tensor<Int32>
+  @noDerivative public var shape: Tensor<Int32>
```
Comment (asuhan) [inline Sources/TensorFlow/Layers/Core.swift:39]: Why was this required?

Comment (pschuh) [reply Sources/TensorFlow/Layers/Core.swift:39]: CopyToDevice requires mutation.
